### DISCUSSION TITLE
Is it ok to use libfork with existing openmp project ?

### DISCUSSION THREAD

**Discussion by Frandy** [ConorWilliams/libfork#7] [Q&A] [answered]
I'd like to try libfork in my project, while I already use openmp and some library such as openblas which also use openmp.
Will it get conflict？

**Comment by ConorWilliams** [comment, marked as answer]
There should be no conflicts however, I would not recommend using one _within_ the other i.e. don't spawn OpenMP regions inside `libfork` coroutines or vice versa as this will spawn too many hardware threads.

Also keep an eye out for the v3 release (you can try the branch out now) this should bring another 2x or more performance.

**Reply by Frandy** [reply]
Thank you very much for quick reply.  It is great.
While I wonder if libfork will keep thread spinning on queue even out of fork region. If so, then it will be conflict with openmp parallel region. But if not, it may suffer thread wakeup overhead.

**Reply by ConorWilliams** [reply]
With V2 the `busy_pool` will sleep all threads when no root tasks are running i.e.

```cpp
busy_pool pool(num_threads);
                                     // All threads are sleeping here.
int fib_10 = pool.schedule(fib(10)); // During the call all threads are awake all the time.
                                     // All threads are back to sleep.
```
In v3 there is another thread pool ``lazy_pool`` which sleeps idle workers when there is not enough work.

**Reply by Frandy** [reply]
Got it. Thank you very much.
